shanoir-ng-datasets/src/main/java/org/shanoir/ng/studycard/controler/StudyCardApiController.java

```diff
@@ -69,6 +69,10 @@ public class StudyCardApiController implements StudyCardApi {
 	public ResponseEntity<Void> deleteStudyCard(
 			@ApiParam(value = "id of the study card", required = true) @PathVariable("studyCardId") Long studyCardId) throws RestServiceException {
 		try {
+			// Will violate integrity constraint
+			if(datasetAcquisitionService.existsByStudyCardId(studyCardId)){
+				return new ResponseEntity<>(HttpStatus.UNPROCESSABLE_ENTITY);
```

Comment:
If you want to return a more precise information (by returning a text msg or anything) you can throw a RestServiceException instead f returning a ResponseEntity. (you can set the UNPROCESSABLE_ENTITY code to the exception)

The exception will be catch elsewhere (GlobalExceptionHandler) and a corresponding ResponseEntity will be build and renturn to the client.